ops/jujuversion.py

```diff
@@ -123,3 +123,9 @@ def has_secrets(self) -> bool:
         # * In 3.0.3, a bug with observer labels was fixed (juju/juju#14916)
         # TODO(benhoyt): update to 3.0.3+ once shipped (for juju/juju#14916 fix)
         return (self.major, self.minor, self.patch) >= (3, 0, 2)
+
+    @property
+    def has_open_port_k8s(self) -> bool:
+        """Report whether this Juju version supports open-port on Kubernetes."""
+        # Support added: https://bugs.launchpad.net/juju/+bug/1920960
+        return (self.major, self.minor, self.patch) >= (3, 0, 3)
```

Comment:
I'm not a huge fan of properties, but these do feel like static information, so I'm more ok with it than most.

That said, the name "has_open_port_k8s" bothers me a bit.  I don't know if I'm missing a word `has_open_port_on_k8s` or `has_open_port_for_k8s`
Or whether I would rather it was `has_open_port` and it figured out if you were on K8s.
Ultimately, I'm happy to have this sort of version check capability, and I'm bikeshedding about the name, but I'd like us to at least think about it a bit.

Reply:
Yeah, fair pushback. I'd rather it not magically figure out whether you were on K8s or not, because `JujuVersion` is a dumb class that just operates on the Juju version, and to take other input (charm metadata or env vars or whatever) would seem against the spirit of what we have here.

However, I've renamed for clarity to include the "on" and renamed "has" to "supports" (because Juju <3.0.3 might "have" open-port on all platforms, it just wouldn't *support* it on K8s). It's now `supports_open_port_on_k8s`, so a bit wordy, but this shouldn't be used often, and clear is better than terse for this.

@carlcsaposs-canonical You happy with `supports_open_port_on_k8s`?

Reply:
LGTM